lib/mongo/protocol/message.rb

```diff
@@ -226,7 +226,7 @@ def serialize_fields(buffer, max_bson_size = nil)
       # @param buffer [String] Buffer to receive the header
       # @return [String] Serialized header
       def serialize_header(buffer)
-        set_request_id unless @request_id
+        set_request_id unless defined?(@request_id)
```

Comment:
Instead of checking if ``@request_id`` is ``#defined?`` can we add something to the ``#set_request_id`` method instead?
Something like the following:

```ruby
      def set_request_id
        @request_id || @@id_lock.synchronize do
          @request_id = @@request_id += 1
        end
      end
```